### PR TITLE
Winduprule 601

### PR DESCRIPTION
**Pull request by johnpoth**
https://issues.redhat.com/browse/WINDUPRULE-601

To run tests:

`mvn -DrunTestsMatching=xml-removed-components clean test` &&
`mvn -DrunTestsMatching=xml-renamed-components clean test`

Thanks !

### CODE REVIEW

**Comment by johnpoth** [comment]
@jonathanvila done! Thanks for reviewing

File: rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml [hunk 1]
```diff
@@ -126,7 +191,20 @@
                 </hint>
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml:191]: Shouldnt it be the other way around with the quickfix ?

File: rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml [hunk 2]
```diff
@@ -156,7 +247,20 @@
...
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml:247]: Shouldn't it be the other way around the quickfix ?

File: rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml [hunk 3]
```diff
@@ -186,7 +303,20 @@
...
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml:303]: Shouldn't it be the other way around the quickfix ?

File: rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml (hunk 1 above)
Comment (johnpoth) [reply rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml:191]: Done, thanks!

File: rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml (hunk 3 above)
Comment (johnpoth) [reply rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml:303]: Done, thanks!

File: rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml (hunk 2 above)
Comment (johnpoth) [reply rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml:247]: Done, thanks!

**Comment by jonathanvila** [comment]
LGTM :+1:

File: rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml [hunk 4]
```diff
@@ -59,10 +85,10 @@
...
             </perform>
             <where param="substitution">
-                <matches pattern="([a-zA-Z0-9]|-)*" />
+                <matches pattern="(?!rxjava2-starter$|http4-starter$|mongodb3-starter$|hdfs2-starter$|hdfs-starter$|mina2-starter$|mina-starter$|netty4-starter$|netty-starter$|netty4-http-starter$|netty-http-starter$|quartz2-starter$|quartz-starter$|jetty9-starter$|aws-starter$)([a-zA-Z0-9]|-)*" />
```
Comment (jonathanvila) [inline rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml:88]: I'm not sure about what is the intention of this regexp .....  
1. you are searching for this in the when &gt;artifact groupId="org.apache.camel" artifactId="camel-{substitution}-starter" &lt; but you are including word starter in the expression  
2. if I understand correctly the expression you say : (not these values at the end of the text) AND (any length of letter or number or dash ) so second group invalidates the first ?

Comment (johnpoth) [reply rules-reviewed/camel3/camel2/xml-renamed-components.windup.xml:88]: @jonathanvila I've just updated the regexp to be a little clearer. Let me know what you think, thanks for reviewing!

**Comment by jonathanvila** [comment]
LGTM :+1:

**Pull request merged**